packages/outputarea/src/model.ts

```diff
@@ -377,31 +377,40 @@ class OutputAreaModel implements IOutputAreaModel {
     }
   }
 
+  /**
+   * Remove characters that are overridden by backspace characters.
+   */
+  private _fixBackspace(txt: string): string {
+    let tmp = txt;
+    do {
+      txt = tmp;
+      // Cancel out anything-but-newline followed by backspace
+      tmp = txt.replace(/[^\n]\x08/gm, '');
```

Comment:
How does this handle `abcd\b\b\b\b`? Does it first match the first `\b`, delete the `d`, then recursively apply to `c\b`? Or does it match `d\b`, then match `\b\b`?

Reply:
Please not that I don't think this handling is correct either. backspace does *not* erase a character. It only moves the insertion point, like \r. Compare:

```
% echo 'a\b'
a
% echo 'a\bc'
c
```
In the first case, I moved the insertion point back to the start of the line, but I didn't overwrite a. In the second case, I moved the insertion point back to the beginning of the line and *also* overwrite `a`.

Reply:
In my testing, it appears to do the latter. It's a little surprising, but also the identical behavior to the classic notebook.

Reply:
Similar issue jupyter/notebook#2892

Reply:
Of course, we may decide we don't want that. Notice what `\r` does here - it only overwrites the single character:

```
% echo 'abc\rA' 
Abc
```
That may be problematic for people thinking that `\r` erases the line.

Reply:
I'm advocating that we go with the classic notebook behavior, and only make intentional changes later if necessary.

Reply:
But to show this regex is the wrong approach, notice this in the js console:

```
"aa\b\b\b".replace(/.[\b]/gm, '')
"a"
```
You see how the `\b\b` was replaced with `''`?

Reply:
I agree with @blink1073 . Especially since libraries have been writing their stdout for things like progress bars to work with the classic notebook.

Reply:
> I'm advocating that we go with the classic notebook behavior, and only make intentional changes later if necessary.

Yes, compatibility (bugs and all) is more important at this point.